### ISSUE TITLE
Status stopped updating, just like the stable tuya integration

### ISSUE THREAD

**Issue by proligde** [tuya/tuya-smart-life#39]
Hi there,

I've moved to this beta version, as the current official tuya (cloud) integration stopped updating after a few hours and needed to be reloaded all the time. I also tried localtuya, but that wasn't an option for me as it didn't support climate sensors - at least not mine.

Now I tried this beta smart-life integration and basically that checks all the boxes - thanks for the great work!

However: It stopped updating again after a few hours, just like the official tuya integration while my phone's smart-life app shows the correct, current data.

Now, I activated debug logging but I'd need some advice about where these logs can be found and which part you would need. Thanks a lot in advance.

**Comment by proligde** [comment]
Update: after restarting HA completely (not only the integration), it started receiving updates again, until this morning at around 04:00:

![image](https://github.com/tuya/tuya-smart-life/assets/998665/e6fbfaef-c9e1-46f5-a09c-1cc057ecfc5a)

The corresponding debug log output around that date is here:

```
2023-11-28 03:37:47.381 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf4e3eb02e1deeb3b0jpza: {'va_temperature': 197, 'va_humidity': 47, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:37:48.874 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf4e3eb02e1deeb3b0jpza: {'va_temperature': 195, 'va_humidity': 47, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:37:49.181 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf4e3eb02e1deeb3b0jpza: {'va_temperature': 195, 'va_humidity': 48, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:37:49.441 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf4e3eb02e1deeb3b0jpza: {'va_temperature': 195, 'va_humidity': 48, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:53:44.221 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bfcca42e2bcb273940pfsu: {'va_temperature': 104, 'va_humidity': 69, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:53:45.992 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bfcca42e2bcb273940pfsu: {'va_temperature': 104, 'va_humidity': 69, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:53:45.996 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bfcca42e2bcb273940pfsu: {'va_temperature': 104, 'va_humidity': 69, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 03:53:46.167 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bfcca42e2bcb273940pfsu: {'va_temperature': 104, 'va_humidity': 69, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 04:01:50.926 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf0fe814bb998a7530enei: {'va_temperature': 206, 'va_humidity': 42, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 04:01:52.564 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf0fe814bb998a7530enei: {'va_temperature': 204, 'va_humidity': 42, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 04:01:53.063 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf0fe814bb998a7530enei: {'va_temperature': 204, 'va_humidity': 42, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 04:01:53.071 DEBUG (Thread-18 (_thread_main)) [custom_components.smartlife] Received update for device bf0fe814bb998a7530enei: {'va_temperature': 204, 'va_humidity': 42, 'battery_state': 'high', 'temp_unit_convert': 'c'}
2023-11-28 04:02:52.995 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': <redacted>, 'expire_time': 7200, 'uid': 'eu1547055791095ekrzV', 'access_token': '<redacted>', 'refresh_token': '<redacted>'}, 'terminal_id': '<redacted>', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-11-28 06:01:53.578 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': <redacted>, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '<redacted>', 'refresh_token': '<redacted>'}, 'terminal_id': '<redacted>', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-11-28 07:11:34.376 DEBUG (SyncWorker_13) [custom_components.smartlife] Sending commands for device 012004092c3ae84480cb: [{'code': <DPCode.SWITCH: 'switch'>, 'value': True}]
2023-11-28 07:11:35.004 DEBUG (SyncWorker_12) [custom_components.smartlife] Sending commands for device 5201703084f3eb20d3c5: [{'code': <DPCode.SWITCH: 'switch'>, 'value': True}]
2023-11-28 07:11:35.238 DEBUG (SyncWorker_1) [custom_components.smartlife] Sending commands for device 012004092c3ae842f72f: [{'code': <DPCode.SWITCH: 'switch'>, 'value': True}]
2023-11-28 07:11:35.495 DEBUG (SyncWorker_9) [custom_components.smartlife] Sending commands for device 012003012c3ae83cd68a: [{'code': <DPCode.SWITCH: 'switch'>, 'value': True}]
2023-11-28 08:00:54.145 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': <redacted>, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '<redacted>', 'refresh_token': '<redacted>'}, 'terminal_id': '<redacted>', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-11-28 09:59:54.571 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': <redacted>, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '<redacted>', 'refresh_token': '<redacted>'}, 'terminal_id': '<redacted>', 'endpoint': 'https://apigw.tuyaeu.com'}
```

So it looks like it is (sucessfully?) retrieving a new token (twice?) and then it stops receiving data.

**Comment by gsqqq123** [comment]
We checked the logs of your device and found that it frequently disconnects from the cloud. 
Is this related to that?

**Comment by proligde** [comment]
@gsqqq123 I have basically two types of devices here:

1. a bunch of climate sensors that run on battery and only connect to the LAN (and this the cloud) a few times per hour. This is intended and seems to be the way these sensors work in order to save power I guess. However they are showing the correct data in the smart life app.

2. Power socket outlets that are always online and should never disconnect. They also work all the time. They just also stop getting their state reported to home assistant (at the same time as the sensors stop reporting)

This means I can switch the sockets on or off with home assistant, however they are stuck on off or on forever although their states are also correctly shown in the smart life app.

So to summarize: the devices and the smart life app on my phone work completely as intended and I'm online 24/7 without any connectivity issues. It's just the integration that shows just the same issues as the classic/official tuya integration.

**Comment by proligde** [comment]
I waited some days, but no change. But everytime I reload the integration I receive a full (Correct) set of sensor data from the smartlife api. So basically the integration would work perfectly fine if I would just restart it every minute but that's obviously not intended :-)

Here is more debug log output (partially shortened) including comments:

```
## no data at all coming in for a few days now - although token update log events seem ok
2023-12-01 09:24:20.783 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701419060772, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '43be53e7e837b15268ce3c1f4eb78d75', 'refresh_token': '76688638dbd10c838131d6f889a02ea0'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-12-01 11:23:21.332 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701426201303, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '91612db70f55edf15d1a3c7b1096d9a9', 'refresh_token': '5901b7dca72b5523b509b85e8c8840b0'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-12-01 13:22:21.866 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701433341853, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': 'cc7adcff78ef216448ad1f58231dbe4c', 'refresh_token': 'fe095c426ddabfcd95bb5e00a126b9a6'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-12-01 15:21:22.417 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701440482403, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '45b9a3d789ba02afd9a9716bce6d16f9', 'refresh_token': '1795d6d36d7a26d3ab90e90653e7105c'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-12-01 17:20:22.992 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701447622969, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': 'f39322d9758b588fbe4320a47bb3b8b7', 'refresh_token': '0ee50149a4ca7414d5cc3cb1b016f5b1'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
2023-12-01 19:19:23.567 DEBUG (Thread-11) [custom_components.smartlife] update token info : {'user_code': '<redacted>', 'token_info': {'t': 1701454763538, 'expire_time': 7200, 'uid': '<redacted>', 'access_token': '826d39ef2631edd4c4a67e4537fa9b00', 'refresh_token': 'bbba70857c4d067a3159599936ec1b27'}, 'terminal_id': '1701011841507HV5c6c', 'endpoint': 'https://apigw.tuyaeu.com'}
## here I manually reloaded the integration
2023-12-01 20:19:24.720 DEBUG (MainThread) [custom_components.smartlife] unload entry id = 2677399f676ba86b62e4dfdce3fa0198
## directly after that I received correct data from the smartlife api again:
2023-12-01 20:19:28.985 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.986 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.986 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.986 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.986 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.986 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.987 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:19:28.988 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
## here I manually reloaded the integration again
2023-12-01 20:20:08.030 DEBUG (MainThread) [custom_components.smartlife] unload entry id = 2677399f676ba86b62e4dfdce3fa0198
## and again - correct data is coming in
2023-12-01 20:20:10.793 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.793 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.794 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.794 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.794 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.794 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.795 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.796 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.796 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.796 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
2023-12-01 20:20:10.796 DEBUG (MainThread) [custom_components.smartlife] dpcode.....
```

Hope this helps. Thanks in advance!

**Comment by shihuimiao** [comment]
Hi there,
Can you check the dependencies of your smartlife, in the file of manifest.json    tuya-device-sharing-sdk==0.1.9  
If it weren't for this version, pull the code again and test it.

**Comment by proligde** [comment]
@shihuimiao thank you! It was indeed only 0.1.8. I updated everything as you suggested and now it reads 0.1.9. I'll report back to this ticket as soon as I see if it's stable now.

**Comment by proligde** [comment]
@shihuimiao unfortunately, the behaviour didn't change. There are still token exchanges in the log but no data is received until I restart the integration or home assistant itself.

![image](https://github.com/tuya/tuya-smart-life/assets/998665/656af6b7-ec2e-47d6-a551-7502ed718ba4)

**Comment by shihuimiao** [comment]
Is your server always online? If it drops, it may cause message loss

**Comment by alanmilinovic** [comment]
I added automation that is reloading integration every 5 minutes.
